### ISSUE TITLE
l1-mixed loss gives NaN for some datasets

### ISSUE THREAD

**Issue by PabloPiaggi** [SparkyTruck/deepmd-jax#27]
I find that the l1-mixed loss gives NaN for some datasets. I think this happens for datasets that have forces that are exactly zero both in the calculation and in the database, for instance configurations with one atom in the simulation box. The l2 loss does not have this problem.

This is an example training script:
```python
from deepmd_jax.train import train

train(
      model_type='energy', 
      rcut=6.0,
      save_path='model.pkl', 
      train_data_path=[
        '../dataset/data.extxyz',
      ], 
      step=2000000, 
      print_loss_smoothing = 1,
      print_every = 1,
      compress=True,
      #loss = "l2",
)
```

and the dataset can be downloaded from this link:
https://drive.google.com/file/d/11bnUb8uUEZtTUEci-QggHPUsbAQeDW9n/view?usp=sharing

**Comment by SparkyTruck** [comment]
This was due to sqrt'(0)=inf → inf·0=NaN in backward pass, now fixed.

**Comment by PabloPiaggi** [comment]
Thanks!
